builtin/providers/softlayer/provider.go

```diff
@@ -0,0 +1,42 @@
+package softlayer
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"username": &schema.Schema{
+				Type: schema.TypeString,
+				Required: true,
+				DefaultFunc: schema.EnvDefaultFunc("SOFTLAYER_USERNAME", nil),
+				Description: "The user name for SoftLayer API operations.",
+			},
+			"api_key": &schema.Schema{
+				Type: schema.TypeString,
+				Required: true,
+				DefaultFunc: schema.EnvDefaultFunc("SOFTLAYER_API_KEY", nil),
+				Description: "The API key for SoftLayer API operations.",
+			},
+		},
+
+		ResourcesMap: map[string]*schema.Resource{
+			"softlayer_virtualserver": resourceSoftLayerVirtualserver(),
```

Comment:
It's actually branded as SoftLayer_Virtual_Guest. We could rename it to `softlayer_virtual_guest`, but that would hurt a lot, since we've already introduced this name in other open source stuff; [microservices-infrastructure](https://github.com/CiscoCloud/microservices-infrastructure/commit/2018cde3a8b867e34f8398578f6f9687240fc7f0) and [terraform.py](https://github.com/CiscoCloud/terraform.py/commit/5a7b1041494a58d331ed79a29ca8146bf272a8d3). (arguably a bit premature :stuck_out_tongue_winking_eye:)